### ISSUE TITLE
[Java] Implement new Concept Exercise: numbers

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1786]
This issue describes how to implement the `numbers` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of the exercise is to understand how to interact with the numbers representation in Java language.

## Learning objectives

- Know of the existence of the two most commonly used number types, `int` and `double`.
- Understand that an `int` represents whole numbers, and a `double` represents floating-point numbers.
- Know of basic operators such as multiplication, comparison and equality.
- Know how to convert from one numeric type to another; know what implicit and explicit conversions are.
- Know how to conditionally execute code using an `if` statement.

## Out of scope

- Any other numeric types besides `int` and `double` (so no `float`, `byte`, etc.).
- Parsing a `string` to an `int` or `double`.
- Converting an `int` or `double` to a `String`.

## Concepts

- `numbers`: know of the existence of the two most commonly used number types, `int` and `double`; understand that the former represents whole numbers, and the latter floating-point numbers; know of basic operators such as multiplication, comparison and equality; know how to convert from one numeric type to another; know what implicit and explicit conversions are.
- `conditionals-if`: know how to conditionally execute code using an `if` statement.

## Prequisites

This exercise's prerequisites Concepts are:

- `basics`: know how to define methods.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing
To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by TalesDias** [comment]
Hey, I would like to do this one

**Comment by mirkoperillo** [comment]
Hi @TalesDias thank you for helping the Java track.
At the moment we are using C# as reference. This is their [exercise](https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers), the alternative is to follow the [freelance rate story](https://github.com/exercism/v3/blob/master/reference/stories/numbers.freelancer-rates.md) implemented by [Javascript](https://github.com/exercism/v3/tree/master/languages/javascript/exercises/concept/numbers) and [Elixir](https://github.com/exercism/v3/tree/master/languages/elixir/exercises/concept/numbers)

I assigned this issue to you, feel free to ask for every doubt

**Comment by mirkoperillo** [comment]
Fixed by #2189
